### PR TITLE
Upgrade pod spec to V3;

### PR DESCRIPTION
**Pull request by ycliuhw**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [ ] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [ ] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [ ] Do comments answer the question of why design decisions were made?

----

## Description of change

Upgrade current podspec to `V3`;

## QA steps

No

## Documentation changes

Yes

## Bug reference

None

### CODE REVIEW

File: caas/kubernetes/provider/specs/v3_test.go [hunk 1]
```diff
@@ -0,0 +1,928 @@
...
+        initialDelaySeconds: 10
+        httpGet:
+          path: /ping
+          port: 8080
+      readinessProbe:
+        initialDelaySeconds: 10
+        httpGet:
+          path: /pingReady
+          port: www
+    config:
```
Comment (wallyworld) [inline caas/kubernetes/provider/specs/v3_test.go]: envConfig

File: caas/kubernetes/provider/specs/v3_test.go [hunk 2]
```diff
@@ -0,0 +1,928 @@
...
+      brackets: '["hello", "world"]'
+      restricted: 'yes'
+      switch: on
+      special: p@ssword's
+      my-resource-limit:
+        resource:
+          container-name: container1
+          resource: requests.cpu
+          divisor: 1m
+    files:
```
Comment (wallyworld) [inline caas/kubernetes/provider/specs/v3_test.go]: fileMounts?
or was it mountPoints?

File: caas/kubernetes/provider/specs/v3_test.go (hunk 1 above)
Comment (ycliuhw) [reply caas/kubernetes/provider/specs/v3_test.go]: As discussed,  we will rename this in #11232

File: caas/kubernetes/provider/specs/v3_test.go (hunk 2 above)
Comment (ycliuhw) [reply caas/kubernetes/provider/specs/v3_test.go]: As discussed,  we will rename this in #11232

**Comment by timClicks** [comment]
Hi @ycliuhw/@wallyworld, sorry if it's obvious.. where are the docs for v3? Also are we renaming it k8s-spec?

**Comment by ycliuhw** [comment]
@timClicks  in this PR, the v3 and v2 are the same.
But we will have some docs when we add changes to v3 later.
Yes, there is a card in todo list for renaming podspec to k8sspec.

**Comment by timClicks** [comment]
Thanks @ycliuhw

**Comment by ycliuhw** [comment]
[podspec V3](https://trello.com/c/hANKYres/1056-podspec-v3)

**Pull request merged**